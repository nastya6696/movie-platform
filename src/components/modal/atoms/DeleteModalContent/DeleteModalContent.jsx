import {Button} from "../../../atoms";

import styles from './styles.module.scss';
import {useContext} from "react";
import {DeleteModalContext} from "../../../molecules/MovieCard/MovieCard";

export const DeleteModalContent = () => {
  const {handleDeleteClick} = useContext(DeleteModalContext);

  return (
    <div className={styles.deleteContent}>
      <p>Are you sure you want to delete this movie?</p>
      <Button name="CONFIRM" handleClick={handleDeleteClick}/>
    </div>
  )
};
