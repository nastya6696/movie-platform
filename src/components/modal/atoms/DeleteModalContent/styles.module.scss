@import "src/assets/scss/common";

.deleteContent {
  color: $white;
  @include fontSizes(20px, 25px);
  margin-top: 37px;

  p {
    margin-bottom: 52px;
  }

  button {
    float: right;
  }
}
