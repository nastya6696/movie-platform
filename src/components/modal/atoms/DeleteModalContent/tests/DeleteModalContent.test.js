import ShallowRenderer from 'react-test-renderer/shallow';
import React from "react";
import {DeleteModalContent} from "../DeleteModalContent";

let realUseContext;
let useContextMock;
// Setup mock
beforeEach(() => {
  realUseContext = React.useContext;
  useContextMock = React.useContext = jest.fn();
});
// Cleanup mock
afterEach(() => {
  React.useContext = realUseContext;
});

jest.mock('../../../../atoms/Button', () => ({
  Button: (({ handleClick, name }) => `
     (--- Button ---): props {
     handleClick: '${handleClick}',
     name: '${name}'
    }`
  )
}));

test("should create simple DeleteModalContent component", () => {
  useContextMock.mockReturnValue(() => {});
  const component = new ShallowRenderer().render(
    <DeleteModalContent />
  );
  expect(component).toMatchSnapshot();
});
