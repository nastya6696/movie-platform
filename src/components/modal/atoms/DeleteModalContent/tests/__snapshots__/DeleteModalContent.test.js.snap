// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mock hook 1`] = `
<div
  className="deleteContent"
>
  <p>
    Are you sure you want to delete this movie?
  </p>
  <Button
    name="CONFIRM"
  />
</div>
`;

exports[`should create simple DeleteModalContent component 1`] = `
<div
  className="deleteContent"
>
  <p>
    Are you sure you want to delete this movie?
  </p>
  <Button
    name="CONFIRM"
  />
</div>
`;
